include/dgl/runtime/device_api.h

```diff
@@ -196,6 +194,15 @@ class DeviceAPI {
    * \return The corresponding device API.
    */
   DGL_DLL static DeviceAPI* Get(DGLContext ctx, bool allow_missing = false);
+
+
+  /*!
+   * \brief Get device API base don context.
+   * \param dev_type The device type
+   * \param allow_missing Whether allow missing
+   * \return The corresponding device API.
+   */
+  DGL_DLL static DeviceAPI* Get(int dev_type, bool allow_missing = false);
```

Comment:
Add this because when pinning an array on CPU, we have to use methods in `CUDADeviceAPI`. With this, we can get the needed device API by an int `device_type`.

Reply:
Use `DLDeviceType` rather than `int`.

Is this function added for convenience so that you don't have to write something like `Get(DLContext{kDLGPU, 0})`?

Reply:
Yes. I think it's clearer to omit the `device_id` in this case, though it doesn't actually matter.